### PR TITLE
Convert Tracing Tests to use Multicluster and Echo App

### PR DESCRIPTION
**Pull request by gargnupur**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ X] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pkg/test/framework/components/opentelemetry/kube.go [hunk 1]
```diff
@@ -31,7 +32,86 @@ type otel struct {
...
+  - name: http-tracing-span
+    number: 55678
+    protocol: http
+  resolution: DNS
+  addresses:
+  # the IP address to which opentelemetry-collector.istio-system.global will resolve to
+  # must be unique for each remote service, within a given cluster.
+  # This address need not be routable. Traffic for this IP will be captured
+  # by the sidecar and routed appropriately.
+  - 240.0.0.2
```
Comment (douglas-reid) [inline pkg/test/framework/components/opentelemetry/kube.go:106]: curious: should this be a non-routable address (ie: 172.16.0.1 or similar)?  how did we select this?

File: tests/integration/telemetry/tracing/tracing.go [hunk 2]
```diff
@@ -18,28 +18,41 @@ package tracing
...
-	bookinfoNsInst namespace.Instance
 	ingInst        ingress.Instance
 	zipkinInst     zipkin.Instance
+	appNsInst      namespace.Instance
+)
+
+const (
+	// For multicluster tests we multiply the number of requests with a
+	// constant multiplier to make sure we have cross cluster traffic.
+	requestCountMultipler = 3
```
Comment (douglas-reid) [inline tests/integration/telemetry/tracing/tracing.go:45]: can we share this in a common place (i think i recall this being used for similar purpose elsewhere) ?

File: pkg/test/framework/components/zipkin/kube.go [hunk 3]
```diff
@@ -37,6 +38,99 @@ const (
...
+  - name: http-tracing-span
+    number: 9411
+    protocol: http
+  resolution: DNS
+  addresses:
+  # the IP address to which zipkin.istio-system.global will resolve to
+  # must be unique for each remote service, within a given cluster.
+  # This address need not be routable. Traffic for this IP will be captured
+  # by the sidecar and routed appropriately.
+  - 240.0.0.2
```
Comment (douglas-reid) [inline pkg/test/framework/components/zipkin/kube.go:125]: same question as above.

File: pkg/test/framework/components/opentelemetry/kube.go (hunk 1 above)
Comment (gargnupur) [reply pkg/test/framework/components/opentelemetry/kube.go:106]: I picked the default one from this example: https://istio.io/latest/docs/setup/install/multicluster/gateways/#configure-the-example-services

File: pkg/test/framework/components/zipkin/kube.go (hunk 3 above)
Comment (gargnupur) [reply pkg/test/framework/components/zipkin/kube.go:125]: Replied above...

File: tests/integration/telemetry/tracing/tracing.go (hunk 2 above)
Comment (gargnupur) [reply tests/integration/telemetry/tracing/tracing.go:45]: Agreed.. Done

**Review by nmittler — APPROVED** [review]

**Review by douglas-reid — APPROVED** [review]

File: prow/integ-suite-kind.sh [hunk 4]
```diff
@@ -96,8 +96,15 @@ while (( "$#" )); do
...
 
-# KinD will not have a LoadBalancer, so we need to disable it
+# Default IP family of the cluster is IPv4
+export IP_FAMILY="${IP_FAMILY:-ipv4}"
+
+# KinD will not have a LoadBalancer, so we need to disable it by default.
 export TEST_ENV=kind
+# LoadBalancer in Kind is supported using metallb if not ipv6.
+if [ "${IP_FAMILY}" != "ipv6" ]; then
+  export TEST_ENV=kind-metallb
```
[the remark this answers is not in the document]
Comment (gargnupur) [reply prow/integ-suite-kind.sh:106]: Thanks..Fixed

**Pull request merged**